extension/src/form-editor/form-editor-provider.ts

```diff
@@ -0,0 +1,86 @@
+import * as vscode from 'vscode';
+import { findRootEntry, parseBuildManifest } from '../base/build-manifest';
+import { calcNonce } from '../base/webview-nonce';
+import { Messenger } from 'vscode-messenger';
+import { FormWebSocketMessage, WebSocketForwarder } from '../websocket-forwarder';
+import { NotificationType } from 'vscode-messenger-common';
+
+const WebviewReadyNotification: NotificationType<void> = { method: 'ready' };
+const InitializeConnectionRequest: NotificationType<{ file: string }> = { method: 'initializeConnection' };
+
+export default class FormEditorProvider implements vscode.CustomTextEditorProvider {
```

Comment:
so it is vscode.CustomTextEditorProvider

Reply:
ah copy paste from the variable editor... hm... need to check that

Reply:
will check it, when I check the dirty state of the editor...